radixdlt/src/main/java/com/radixdlt/consensus/validators/ValidationState.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * (C) Copyright 2020 Radix DLT Ltd
+ *
+ * Radix DLT Ltd licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied.  See the License for the specific
+ * language governing permissions and limitations under the License.
+ */
+
+package com.radixdlt.consensus.validators;
+
+import java.util.Objects;
+import java.util.concurrent.ConcurrentMap;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Maps;
+import com.radixdlt.crypto.ECDSASignature;
+import com.radixdlt.crypto.ECDSASignatures;
+import com.radixdlt.crypto.ECPublicKey;
+import com.radixdlt.crypto.Hash;
+
+/**
+ * Keeps track of current validation state for a thing that
+ * needs multiple correct signatures for a quorum.
+ */
+public final class ValidationState {
```

Comment:
It feels a bit odd to conflate the requirements for a quorum (`acceptableFaults` and all that) with the state used to keep track of signatures, checking against those requirements. Probably fine for now, just wanted to flag.

Reply:
Could rename to QuorumState, I guess.  Not sure it really makes sense to break the thresholding out into something separate right now, although I feel kind of sure that this is going to be subject to some change in the future in any case.